src/App.jsx

```diff
@@ -14,33 +16,145 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const buttonsFromServer = [
+  'Sort alphabetically',
+  'Sort by length',
+  'Reverse',
+  'Reset',
+];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+function getPreparedButtons(buttons, { sortField, reverse }) {
+  const preparedButtons = [...buttons];
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (!sortField && !reverse) {
+    preparedButtons.pop();
+  }
+
+  return preparedButtons;
+}
+
+function getPreparedGoods(goods, { sortField, reverse }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField === SORT_FIELD_NAME) {
+        return good1.localeCompare(good2);
+      }
+
+      // if (sortField === SORT_FIELD_NAME && reverse === true) {
+      //   return good2.localeCompare(good1);
+      // }
```

Comment:
The commented-out code for reverse sorting by name is not implemented. If reverse sorting by name is required, consider implementing this logic.